File: 14.java
import java.util.*;
import java.lang.*;
import java.io.*;

class GFG {
	public static void main (String[] args) {
		//code
		Scanner sc = new Scanner(System.in);
		int cases = sc.nextInt();
		while(cases-->0){
		    int size = sc.nextInt();
		    int[] a = new int[size];
		    for(int i=0;i<size-1;i++){
		        int n = sc.nextInt();
		        a[n-1] = n;
		    }
		    for(int i=0;i<size;i++)
		        if(a[i]!=i+1){
		            System.out.println(i+1);
		            break;
		        }
		}
	}
}

https://practice.geeksforgeeks.org/problems/missing-number/0/?ref=self

File: 14.c
#include <stdio.h>

int main() {
	//code
	int cases;
	scanf("%d",&cases);
	while(cases-->0){
	    int size;
	    scanf("%d",&size);
	    int a[size];
	    scanf("%d",&a[0]);
	    for(int i=1;i<size;i++){
	        scanf("%d",&a[i]);
	        if(a[i]<a[i-1])printf("%d ",a[i]);
	        else printf("-1 ");
	    }
	    printf("-1\n");
	}
	return 0;
}

https://practice.geeksforgeeks.org/problems/immediate-smaller-element/0/?ref=self
